python/lsst/pipe/analysis/coaddAnalysis.py

```diff
@@ -477,16 +491,29 @@ def readSrcMatches(self, dataRefList, dataset, hscRun=None, wcs=None, aliasDictL
             # Set some aliases for differing schema naming conventions
             if aliasDictList:
                 catalog = setAliasMaps(catalog, aliasDictList)
-            if dataset != "deepCoadd_meas" and any(ss not in catalog.schema
-                                                   for ss in self.config.columnsToCopy):
-                unforced = dataRef.get("deepCoadd_meas", immediate=True,
+            if dataset != (self.config.coaddName + "Coadd_meas" and
+                           any(ss not in catalog.schema for ss in list(self.config.columnsToCopyFromMeas) +
+                               list(self.config.columnsToCopyFromRef))):
```

Comment:
I don't think these need to be cast as lists either.

Reply:
If I don't cast I get:
`TypeError: can only concatenate list (not "List") to list`
(I guess `lsst.pex.config` `ListFields` have a different effective `type`).

I defined a `needToAddColumns` variable to make this clearer.  The logic was also slightly flawed, so that got fixed too (so thanks for the request to clarify!)